source/extensions/access_loggers/grpc/grpc_access_log_impl.cc

```diff
@@ -38,7 +40,24 @@ GrpcAccessLoggerImpl::GrpcAccessLoggerImpl(Grpc::RawAsyncClientPtr&& client, std
   flush_timer_->enableTimer(buffer_flush_interval_msec_);
 }
 
+bool GrpcAccessLoggerImpl::canLogMore() {
+  if (buffer_size_bytes_ == 0 || approximate_message_size_bytes_ < buffer_size_bytes_) {
```

Comment:
not your code but do you mind doing s/buffer_size_bytes_/max_buffer_size_bytes_ ? I had to go look this up.